src/framework/plugin.ts

```diff
@@ -40,9 +40,37 @@ export type WorkflowHooks = {
   dev: {
     onStart?: SideEffector
     onFileWatcherEvent?: Chokidar.FileWatcherEventCallback
-    addToSettings: {
+    addToWatcherSettings: {
+      /**
+       * Set additional files to be watched for the core and plugin listeners
+       */
       watchFilePatterns?: string[]
```

Comment:
Feels like simplest thing we can do is keep the symmetry of `watchFilePatterns` aligned with `ignoreFilePatterns`. Plugin author can stay DRY with variables on their side right?

Reply:
So, there's a fundamental difference between `watchFilePatterns` and `plugin.allowFilePatterns` as explained by the doc. `watchFilePatterns` adds files to be watched, which can then be ignored or allowed by the core and plugins listeners.

The reason why it's there and not in `core.watchFilePatterns` is because adding file to that property *will* trigger both the core and plugin listeners unless set otherwise (with `allow/ignoreFilePatterns`)